smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaThrottledLatestProcessedCommit.java

```diff
@@ -196,7 +196,7 @@ private void startFlushAndCheckHealthTimer() {
         Uni<OffsetStore> uni;
         if (offsetStore == null) {
             uni = consumer.committed(recordsTopicPartition)
-                    .emitOn(runnable -> context.runOnContext(x -> runnable.run())) // Switch back to event loop
+                    .emitOn(runnable -> runOnRecordContext(record, runnable)) // Switch back to event loop
```

Comment:
Here we need to go back to the event loop, but you go back to the record context.